tests/integration/commands/declare_command_test.py

```diff
@@ -26,6 +34,67 @@ def compiled_contract_path_fixture() -> Path:
     return Path("./build/main.json")
 
 
+async def test_declaring_cairo1_contract(
+    devnet_gateway_url: str,
+    devnet_account: DevnetAccount,
+    set_private_key_env_var: SetPrivateKeyEnvVarFixture,
+    datadir: Path,
+):
+    class MockedProjectCompiler(ProjectCompiler):
+        def __init__(self):
+            super().__init__(MagicMock(), MagicMock())
+
+        def compile_contract_to_sierra_from_contract_name(
+            self, *args: Any, **kwargs: Any
+        ):
+            # pylint: disable=unused-argument
+            compiled = cairo_bindings.compile_starknet_contract_to_sierra_from_path(
+                input_path=datadir / "minimal.cairo",
```

Comment:
For whatever reason declaring more complicated contracts fails on devnet i.e. when I replace `minimal.cairo` with `balance.cairo` it fails.

Trying to figure out if this is devet, protostar, or starknet.py issue.

Reply:
Turned out it was local setup problem. Devnet has to have a correct path to cairo1 compiler main Cargo.toml set up for declare cairo1 contracts to work correctly.